lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2148,14 +2149,51 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
         continue;
       }
 
-      // Tensor operand type's are ok.
+      // Handle TensorFlow values. These must be handled separately from
+      // aggregates of TensorFlow values (handled below), so that we do not
+      // generate input lists for them.
       if (isTensorFlowValue(operandTy)) {
```

Comment:
Please also change the non-list cases to auto-unwrap, when the aggregate has exactly one terminal element.

After this is done, all the hacky AST-level unpacking code can be nuked! https://github.com/apple/swift/blob/ed4eab5bd272192c5a44f2de7aa24db49472368d/lib/Sema/CSApply.cpp#L2477 (lines 2477-2527).